src/index.ts

```diff
@@ -51,34 +53,53 @@ function bad(message: string): APIGatewayProxyResult {
   };
 }
 
+function serviceUnavailable(message: string): APIGatewayProxyResult {
+  return {
+    statusCode: 503,
+    headers: {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    },
+    body: JSON.stringify({response: message})
+  };
+}
+
 const handler: APIGatewayProxyHandler =
     (event: APIGatewayProxyEvent, context: Context,
      callback: APIGatewayProxyCallback) => {
-      // make consent record
-      const consentRecord = new ConsentRecord();
-      if (event.body != null) {
-        const bodyJson = JSON.parse(event.body);
-        consentRecord.browserId = bodyJson['browser_id'];
-        consentRecord.consentStr = bodyJson['consent_str'];
+      if (STREAM_NAME) {
```

Comment:
I guess this is the bit I'm talking about above. It's a *SUPER* minor point though, especially because the nature of typescript means that within this `if` block the undefined nature has gone away.

Reply:
Yeah, could just compare to a MISSING_VALUE or something similar.